src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQuery.java

```diff
@@ -870,5 +874,31 @@ public static DataTypeDefinition getDataTypeDefinition(DictionaryService diction
             dbids[i] = getDbid(stringValues[i], nodeDAO, tenantService);
         }
         return dbids;
-    }
+    }
+
+    public int getMaxItems()
+    {
+        return maxItems;
+    }
+    
+    public int getSkipCount()
+    {
+        return skipCount;
+    }
+
+    public void setMaxItems(int maxItems)
+    {
+        this.maxItems = maxItems;
+    }
+
+    public void setSkipCount(int skipCount)
+    {
+        this.skipCount = skipCount;
+    }
+    
+    public int getQueryLimit() 
+    {
+        return Math.max(1001, maxItems);
```

Comment:
Limiting the DB result to 1000 won't work in cases where the user doesn't have permission to view the result. So, this means the pagination is broken for the client. Note that permission check is performed after DB fetch.
@skopf what do think about this limitation?

Reply:
The PermissionService limits the number of checks to a maximum of 1000 anyway, probably populated through configuration. I guess we can inject here the same value to have consistency.

Reply:
Precisely here:
https://github.com/Alfresco/alfresco-repository/blob/283beb2d38a6be81089c78a79e3a9fb180d6d635/src/main/java/org/alfresco/repo/security/permissions/impl/acegi/ACLEntryAfterInvocationProvider.java#L624-L630

Reply:
As of this [commit](https://github.com/Alfresco/alfresco-repository/pull/1187/commits/5bb3fa6ea0559f76754efe4a25a6c6797a530ce1) the query limit is set through the pre-configured value of `maxPermissionChecks` found in the repository properties 
https://github.com/Alfresco/alfresco-repository/blob/283beb2d38a6be81089c78a79e3a9fb180d6d635/src/main/resources/alfresco/repository.properties#L154-L155